web/src/js/ducks/ui/optionsEditor.ts

```diff
@@ -15,68 +12,44 @@ type OptionsEditorState = Partial<{
     [name in Option]: OptionUpdate<OptionsState[name]>;
 }>;
 
-const defaultState: OptionsEditorState = {};
-
-export default function reducer(state = defaultState, action) {
-    switch (action.type) {
-        case OPTION_UPDATE_START:
-            return {
-                ...state,
-                [action.option]: {
-                    isUpdating: true,
-                    value: action.value,
-                    error: false,
-                },
-            };
-
-        case OPTION_UPDATE_SUCCESS:
-            return {
-                ...state,
-                [action.option]: undefined,
+export const defaultState: OptionsEditorState = {};
+
+const optionsEditorSlice = createSlice({
+    name: "ui/optionsEditor",
+    initialState: defaultState,
+    reducers: {
+        startUpdate(state, action: PayloadAction<{ option: any; value: any }>) {
```

Comment:
Unfortunately, we cannot use `Partial<{[name in Option]: OptionsState[name]}>` because TypeScript complains that the index can be undefined, and it cannot be used as an index type. The error is as follows:

`Type 'Partial<{ add_upstream_certs_to_client_chain: boolean; allow_hosts: string[]; anticache: boolean; anticomp: boolean; block_global: boolean; block_list: string[]; block_private: boolean; body_size_limit: string | undefined; ... 89 more ...; websocket: boolean; }>' cannot be used as an index type.`

Even when I try to manually check if it's not undefined, it still doesn't work.

Reply:
As far as I’ve seen in RTK, when we have more than one parameter, we need to use an object in the `PayloadAction`. If we want to maintain the previous API, like `startUpdate(key, val)`, I can define an helper function in the slice to map the parameters into an object. What do you think?